File: src/main/java/com/deneb/client/utils/ColorUtil.java
package com.deneb.client.utils;

import java.awt.*;

public class ColorUtil {
    public static Color getColor(int hex){
        return new Color(hex);
    }
    public static int getRed(int hex){
        return hex >> 16 & 255;
    }
    public static int getGreen(int hex){
        return hex >> 8 & 255;
    }
    public static int getBlue(int hex){
        return hex & 255;
    }

    public static int getHoovered(int color,boolean isHoovered){
        return isHoovered ? (color & 0x7F7F7F) << 1 : color;
    }

    public static int getHoovered(Color color,boolean isHoovered){
        return isHoovered ? (color.getRGB() & 0x7F7F7F) << 1 : color.getRGB();
    }
}

File: src/main/java/club/deneb/client/utils/ColorUtil.kt
package club.deneb.client.utils

import java.awt.Color

object ColorUtil {
    fun getColor(hex: Int): Color {
        return Color(hex)
    }

    fun getRed(hex: Int): Int {
        return hex shr 16 and 255
    }

    fun getGreen(hex: Int): Int {
        return hex shr 8 and 255
    }

    fun getBlue(hex: Int): Int {
        return hex and 255
    }

    @JvmStatic
    fun getHoovered(color: Int, isHoovered: Boolean): Int {
        return if (isHoovered) color and 0x7F7F7F shl 1 else color
    }

}